### PR TITLE
Workaround Kotlin lack of Java 21 support in `kotlin-dsl` plugin

### PR DESCRIPTION
**Pull request by eskatos**
By capping the JVM target to the last version supported by Kotlin.

* Part of https://github.com/gradle/gradle/issues/25574

### CODE REVIEW

File: platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt [hunk 1]
```diff
@@ -59,4 +73,48 @@ abstract class KotlinDslPlugin : Plugin<Project> {
...
+                targetJavaVersion.map { JvmTarget.fromTarget(it.toString()) }
+            )
+        }
+        tasks.withType(JavaCompile::class.java).configureEach { javaCompile ->
+            javaCompile.targetCompatibility = targetJavaVersion.get().toString()
+        }
+    }
+
+    private
+    val Project.deprecatedJvmTargetVersion: Provider<JavaVersion>
```
Comment (jbartok) [inline platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt:93]: I would say don't call this "deprecated...", it's very confusing. I now know what you mean, but took me a while.

**Review by jbartok — APPROVED** [review]
LGTM, just a minor comment on naming.

File: platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt (hunk 1 above)
Comment (eskatos) [reply platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt]: Renamed

**Comment by eskatos** [comment]
@bot-gradle test QFL

**Comment by eskatos** [comment]
* Superseded by https://github.com/gradle/gradle/pull/26758

**Pull request closed without merging**